clang/test/CodeGenHLSL/static-local-ctor.hlsl

```diff
@@ -0,0 +1,37 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.0-compute -emit-llvm -o - -disable-llvm-passes %s | FileCheck %s
+
+// Verify that no per variable _Init_thread instructions are emitted for non-trivial static locals
+// These would normally be emitted by the MicrosoftCXXABI, but the DirectX backend should exlude them
+// Instead, check for the guardvar oparations that should protect the constructor initialization should
+// only take place once.
+
+RWBuffer<int> buf[10];
+
+void InitBuf(RWBuffer<int> buf) {
+  for (unsigned i; i < 100; i++)
+    buf[i] = 0;
+}
+
+// CHECK-NOT: _Init_thread_epoch
+// CHECK: define internal void @"?main@@YAXXZ"
+// CHECK-NEXT: entry:
+// CHECK-NEXT: [[Tmp1:%.*]] = alloca %"class.hlsl::RWBuffer"
+// CHECK-NEXT: [[Tmp2:%.*]] = load i32, ptr
+// CHECK-NEXT: [[Tmp3:%.*]] = and i32 [[Tmp2]], 1
+// CHECK-NEXT: [[Tmp4:%.*]] = icmp eq i32 [[Tmp3]], 0
+// CHECK-NEXT: br i1 [[Tmp4]]
+// CHECK-NOT: _Init_thread_header
+// CHECK: init:
+// CHECK-NEXT: = or i32 [[Tmp2]], 1
+// CHECK-NOT: _Init_thread_footer
+
+
+[shader("compute")]
+[numthreads(1,1,1)]
+void main() {
+  // A non-trivially initialized static local will get checks to verify that it is generated just once
+  static RWBuffer<int> mybuf;
+  mybuf = buf[0];
```

Comment:
That's what I'm expecting and that's fine. RWBuffer has an implicit constructor that needs the guards. It's little different if I initialized it on the same line as far as the guard variable protection goes. I'm using "initialization" a bit more broadly than in the C++ context because that's what the compiler code uses to refer to the construction execution. 

You can see the calls I'm eliminating here in this link: https://godbolt.org/z/d7djPTWP7
```llvm
tail call void @_Init_thread_header(ptr nonnull @"?$TSS0@?1??main@@YAXXZ@4HA") #1, !dbg !75
...
  br i1 %26, label %27, label %31, !dbg !75

27:                                               ; preds = %24
...
  tail call void @_Init_thread_footer(ptr nonnull @"?$TSS0@?1??main@@YAXXZ@4HA") #1, !dbg !75
```